app/javascript/mastodon/reducers/compose.js

```diff
@@ -261,7 +261,6 @@ export default function compose(state = initialState, action) {
     });
   case COMPOSE_SPOILERNESS_CHANGE:
     return state.withMutations(map => {
-      map.set('spoiler_text', '');
```

Comment:
But if you don't clear this, the param will be submitted to the API even if it's not displayed in the UI and the post will be created with an unexpected CW?

Reply:
No. There's also `spoiler` which controls whether the spoiler field is shown or not, and it's correctly looked at at submitting time.